edflow/hooks/logging_hooks/tf_logging_hook.py

```diff
@@ -76,9 +86,71 @@ def after_step(self, batch_index, last_results):
             for name in sorted(logs.keys()):
                 self.logger.info("{}: {}".format(name, logs[name]))
 
-            for name, image_batch in last_results["images"].items():
-                full_name = name + "_{:07}.png".format(step)
-                save_path = os.path.join(self.root, full_name)
-                plot_batch(image_batch, save_path)
+            if not self.log_images_to_tensorboard:
+                for name, image_batch in last_results["images"].items():
+                    full_name = name + "_{:07}.png".format(step)
+                    save_path = os.path.join(self.root, full_name)
+                    plot_batch(image_batch, save_path)
 
             self.logger.info("project root: {}".format(self.root))
+
+
+class ImageOverviewHook(Hook):
+    def __init__(self, images={}, interval=100, root_path="logs"):
+        """
+        Logs an overview of all image outputs at an intervall of training steps.
+        Args:
+            scalars (dict): Scalar ops.
+            histograms (dict): Histogram ops.
+            images (dict): Image ops. Note that for these no
+                tensorboard logging ist used but a custom image saver.
+            logs (dict): Logs to std out via logger.
+            graph (tf.Graph): Current graph.
+            interval (int): Intervall of training steps before logging.
+            root_path (str): Path at which the logs are stored.
+        """
+
+        summary_op = tf.no_op()
+        # self.log_images_to_tensorboard = log_images_to_tensorboard
+        # TODO: actually implement this functionality
+
+        self.fetch_dict = {"summaries": summary_op, "images": images}
+
+        self.interval = interval
+        self.root = root_path
+        self.logger = get_logger(self)
+
+    def before_epoch(self, ep):
+        pass
+
+    def before_step(self, batch_index, fetches, feeds, batch):
+        if batch_index % self.interval == 0:
```

Comment:
not a big deal but can we just `pass` without this unnecessary `if`? Or better yet, aren't the default implementations of `before_epoch`, `before_step` to do nothing anyway? just checked---they are. you can just remove lines 123-128